cpp/src/io/parquet/reader_impl.cu

```diff
@@ -629,6 +635,26 @@ class aggregate_metadata {
         if (schema_elem.num_children == 0) {
           input_column_info& input_col =
             input_columns.emplace_back(input_column_info{schema_idx, schema_elem.name});
+
+          // set up child output column for one-level encoding list
+          if (schema_elem.is_one_level_list()) {
+            // determine the element data type
+            auto const element_type = to_type_id(
+              schema_elem, strings_to_categorical, timestamp_type_id, strict_decimal_types);
+            auto const element_dtype =
+              element_type == type_id::DECIMAL32 || element_type == type_id::DECIMAL64
+                ? data_type{element_type, numeric::scale_type{-schema_elem.decimal_scale}}
+                : data_type{element_type};
```

Comment:
This block repeats now, might be good to add `type_id_to_dtype` (or to_data_type), that generates `data_type` from `type_id` and `SchemaElement`.
Also, maybe `data_type` should have a ctor that ignores scale if the type is not a fixed point type, but that's a wider topic.